Assets/LeapMotion/Core/Scripts/HandModelManager.cs

```diff
@@ -13,37 +13,159 @@
 using System.Collections.Generic;
 using UnityEngine.Serialization;
 using System;
+using Leap.Unity.Attributes;
 #if UNITY_EDITOR
 using UnityEditor;
 #endif
-using Leap;
 
 namespace Leap.Unity {
-  //
-  /**
-   * HandPool holds a pool of HandModelBases and makes HandRepresentations
-   * when given a Leap Hand and a model type of graphics or physics.
-   * When a HandRepresentation is created, a HandModelBase is removed from the pool.
-   * When a HandRepresentation is finished, its HandModelBase is returned to the pool.
-   */
-  public class HandPool : MonoBehaviour {
+
+  /// <summary>
+  /// The HandModelManager manages a pool of HandModelBases and makes HandRepresentations
+  /// when a it detects a Leap Hand from its configured LeapProvider.
+  /// 
+  /// When a HandRepresentation is created, a HandModelBase is removed from the pool.
+  /// When a HandRepresentation is finished, its HandModelBase is returned to the pool.
+  /// 
+  /// This class was formerly known as HandPool.
+  /// </summary>
+  public class HandModelManager : MonoBehaviour {
+    
+    #region Formerly in LeapHandController
+
+    protected LeapProvider provider;
```

Comment:
This is a duplicate reference to a provider, there is already one (in the inspector too) named _leapProvider.